### PR TITLE
Fixes part of #935: Optimised test cases for HDPI devices for remaining packages

### PR DESCRIPTION
**Pull request by NullByte08**
## Explanation
Fixes part of #935 :
Tests made compatible for `HDPI` devices for all the remaining packages. All the tests pass successfully.

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt [hunk 1]
```diff
@@ -344,16 +372,80 @@ class NavigationDrawerTestActivityTest {
...
           R.id.home_recycler_view,
           0,
           R.id.welcome_text_view
         )
-      ).check(matches(withText("Welcome to Oppia!")))
+      ).check(matches(withText("Good morning,")))
+    }
+  }
+
+  fun nestedScrollTo(): ViewAction {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt]: Please update all these functions as mentioned on other PR review.

File: app/src/sharedTest/java/org/oppia/app/topic/revision/TopicRevisionFragmentTest.kt [hunk 2]
```diff
@@ -167,6 +172,35 @@ class TopicRevisionFragmentTest {
     return ActivityScenario.launch(intent)
   }
 
+  class GridLayoutManagerColumnCountAssertion(expectedColumnCount:Int): ViewAssertion {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/topic/revision/TopicRevisionFragmentTest.kt]: Update on suggested optimisation:
```
 class GridLayoutManagerColumnCountAssertion(expectedColumnCount: Int) : ViewAssertion {
    private var expectedColumnCount: Int = 0

    init {
      this.expectedColumnCount = expectedColumnCount
    }

    override fun check(view: View, noViewFoundException: NoMatchingViewException?) {
      if (noViewFoundException != null) {
        throw noViewFoundException
      }
      val recyclerView = view as RecyclerView
      if (recyclerView.layoutManager is GridLayoutManager) {
        val gridLayoutManager = recyclerView.layoutManager as GridLayoutManager
        val spanCount = gridLayoutManager.spanCount
        if (spanCount != expectedColumnCount) {
          val errorMessage =
            ("expected column count " + expectedColumnCount + " but was " + spanCount)
          throw AssertionError(errorMessage)
        }
      } else {
        throw IllegalStateException("no grid layout manager")
      }
    }
  }
```

**Review by rt4914 — CHANGES_REQUESTED** [review]
All tests are passing, suggested optimmisations.

**Comment by NullByte08** [comment]
@rt4914 Changes done.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Tests are failing on circle CI because one import is missing in  `NavigationDrawerTestActivityTest`, once you fix that these test cases should pass. Also, do merge it with latest develop just to be on safe side.

**Comment by NullByte08** [comment]
@rt4914 Changes done. Please merge. Thank you.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**